pkg/process/config/yaml_config.go

```diff
@@ -57,31 +56,12 @@ func (a *AgentConfig) LoadProcessYamlConfig(path string) error {
 		a.HostName = config.Datadog.GetString("hostname")
 	}
 
-	// Note: The enabled environment flag operates differently than that of our YAML configuration
-	if v, ok := os.LookupEnv("DD_PROCESS_AGENT_ENABLED"); ok {
-		// DD_PROCESS_AGENT_ENABLED: true - Process + Container checks enabled
-		//                           false - No checks enabled
-		//                           (none) - Container check enabled (by default)
-		if enabled, err := isAffirmative(v); enabled {
-			a.Enabled = true
-			a.EnabledChecks = processChecks
-		} else if !enabled && err == nil {
-			a.Enabled = false
-		}
-	} else if k := key(ns, "enabled"); config.Datadog.IsSet(k) {
-		// A string indicate the enabled state of the Agent.
-		//   If "false" (the default) we will only collect containers.
-		//   If "true" we will collect containers and processes.
-		//   If "disabled" the agent will be disabled altogether and won't start.
-		enabled := config.Datadog.GetString(k)
-		ok, err := isAffirmative(enabled)
-		if ok {
-			a.Enabled, a.EnabledChecks = true, processChecks
-		} else if enabled == "disabled" {
-			a.Enabled = false
-		} else if !ok && err == nil {
-			a.Enabled, a.EnabledChecks = true, containerChecks
-		}
+	a.Enabled = false
+	if config.GetProcessCollectionEnabled(config.Datadog) {
+		a.Enabled, a.EnabledChecks = true, processChecks
+	}
+	if config.GetContainerCollectionEnabled(config.Datadog) {
```

Comment:
This is the spot where I would like the earlier comment moved - you just need to add `else` and the logic of handling these sets of checks should be correct:
```suggestion
	if config.GetProcessCollectionEnabled(config.Datadog) {
		a.Enabled, a.EnabledChecks = true, processChecks
	} else if config.GetContainerCollectionEnabled(config.Datadog) {
		// Container checks are enabled only when process checks are not (since they automatically collect container data).
```

Reply:
Makes sense, thanks for the suggestion!